tslearn/metrics.py

```diff
@@ -2031,6 +2039,241 @@ def cdist_soft_dtw_normalized(dataset1, dataset2=None, gamma=1.):
     return dists
 
 
+@njit(nogil=True)
+def njit_lcss_accumulated_matrix(s1, s2, eps, delta):
+    """Compute the longest common subsequence similarity score between
+    two time series.
+
+    Parameters
+    ----------
+    s1 : array, shape = (sz1,)
+        First time series.
+
+    s2 : array, shape = (sz2,)
+        Second time series.
+
+    eps : float
+        Matching threshold.
+
+    delta : int
+        Maximum distance in the x-axis (time) for the matching to occur.
+
+    Returns
+    -------
+    lcss_score : float
+        Longest Common Subsequence similarity score between both time series.
+
+    """
+    l1 = s1.shape[0]
+    l2 = s2.shape[0]
+    acc_cost_mat = numpy.full((l1 + 1, l2 + 1), 0)
+
+    for i in range(1, l1 + 1):
+        for j in range(1, l2 + 1):
+            if _local_eucl_dist(s1[i - 1], s2[j - 1]) <= eps\
+                    and abs(i - j) <= delta:
+                acc_cost_mat[i][j] = 1 + acc_cost_mat[i - 1][j - 1]
+            else:
+                acc_cost_mat[i][j] = max(acc_cost_mat[i][j - 1],
+                                         acc_cost_mat[i - 1][j])
+
+    return acc_cost_mat
+
+
+@njit(nogil=True)
+def njit_lcss(s1, s2, eps, delta):
+    """Compute the longest common subsequence score between two time series.
+
+    Parameters
+    ----------
+    s1 : array, shape = (sz1,)
+        First time series.
+
+    s2 : array, shape = (sz2,)
+        Second time series
+
+    eps : float (default: 1.)
+        Maximum matching distance threshold.
+
+    delta : int (default: inf)
```

Comment:
This delta parameter looks a lot like the Sakoe-Chiba band width. Maybe using the same notations in both cases would help the end-user see that

Reply:
> I'm not sure how important it can be to implement path constraints for it. 

So you do think I should remove that from LCSS?